### ISSUE TITLE
Cannot use `-` on staged or untracked files in `:Gstatus`

### ISSUE THREAD

**Issue by TSFoster** [tpope/vim-fugitive#1083]
Hi,

Hitting `-` on staged or untracked files does nothing in `:Gstatus` (works fine for tracked files with changes). I'm running git v2.18.0 on MacOS 10.13.6. `dp` also doesn't work on untracked files.

This happened when using Nvim v0.3.1/fish v2.7.1, but the same thing happens when using Vim v8.0/bash v3.2.57 and with a minimal vimrc (pathogen + fugitive).

**Comment by tpope** [comment]
Half working is odd. Sounds like maybe an issue parsing out the sections. Please share the entire contents of a `:Gstatus` buffer.

**Comment by TSFoster** [comment]
```
# On branch master
# Changes to be committed:
#   (use "git reset HEAD <file>..." to unstage)
#
#	new file:   b.txt
#
# Changes not staged for commit:
#   (use "git add <file>..." to update what will be committed)
#   (use "git checkout -- <file>..." to discard changes in working directory)
#
#	modified:   a.txt
#
# Untracked files:
#   (use "git add <file>..." to include in what will be committed)
#
#	c.txt
#
```

Thanks for your help

**Comment by tpope** [comment]
Can you confirm this if this is a failure to stage/unstage the file, or if it's just a redraw issue? That is, if you run `git status` in the terminal, does it show the operation was actually successful?

Untracked files are added with `git add -A ./filename` and staged files are reset with `git reset -q ./filename`. Can you verify those both work?

What version of Git are you running?

**Comment by tpope** [comment]
Also try both 45c21fc38411da91f25978ce6473a9592a8b05dc and the commit before it, as that's a likely culprit.

**Comment by TSFoster** [comment]
Yup, 45c21fc is the commit that breaks it! Not a redraw issue. git v2.18.0. Impressive debugging! Anything else I can do?

**Comment by tpope** [comment]
Verify those 2 commands in the second paragraph of my last comment work.

**Comment by tpope** [comment]
You can also change `echo` to `throw` here and see if there's an error that's not displaying for some reason: https://github.com/tpope/vim-fugitive/blob/fb5531997342be62efbbb98bda4b90ee88b2ac00/autoload/fugitive.vim#L1911
